Modules/_sqlite/connection.c

```diff
@@ -1106,6 +1112,9 @@ int pysqlite_check_thread(pysqlite_Connection* self)
 
 static PyObject* pysqlite_connection_get_isolation_level(pysqlite_Connection* self, void* unused)
 {
+    if (!self || !self->isolation_level) {
+        Py_RETURN_NONE;
```

Comment:
The other two getters raise a `ProgrammingError` in case the `Connection` object is not initialized, so ISTM that this one should behave the same..

Reply:
First of all, thanks for your comments.
I'm working on the next commit with all the suggested fixes and (hopefully) some added tests.

Reply:
don't worry, it's cool